java/client/src/main/java/glide/api/commands/BitmapBaseCommands.java

```diff
@@ -95,4 +95,95 @@ public interface BitmapBaseCommands {
+     *     returned.
+     * @example
+     *     <pre>{@code
+     * Long payload = client.bitpos("myKey1", 0).get();
+     * assert payload == 3L; // Indicates that the first occurrence of a 0 bit value of the string stored at "myKey1" is at the fourth position.
+     * }</pre>
+     */
+    CompletableFuture<Long> bitpos(String key, long bit);
+
+    /**
+     * Return the position of the first bit matching the given <code>bit</code> value. The offset
+     * <code>start</code> is a zero-based index, with <code>0</code> being the first byte of the list,
+     * <code>1</code> being the next byte and so on. These offsets can also be negative numbers
+     * indicating offsets starting at the end of the list, with <code>-1</code> being the last byte of
+     * the list, <code>-2</code> being the penultimate, and so on.
+     *
+     * @see <a href="https://redis.io/commands/bitpos/">redis.io</a> for details.
+     * @param key The key of the string.
+     * @param bit The bit value to match.
+     * @param start The starting offset.
+     * @return The position of the first occurrence beginning at the <code>start</code> offset of the
+     *     <code>bit</code> in the binary value of the string held at <code>key</code>.
+     * @example
+     *     <pre>{@code
+     * Long payload = client.bitpos("myKey1", 1, 4).get();
+     * assert payload == 9L; // Indicates that the first occurrence of a 1 bit value starting from fifth byte of the string stored at "myKey1" is at the tenth position.
+     * }</pre>
+     */
+    CompletableFuture<Long> bitpos(String key, long bit, long start);
+
+    /**
+     * Return the position of the first bit matching the given <code>bit</code> value. The offsets
+     * <code>start</code> and <code>end</code> are zero-based indexes, with <code>0</code> being the
+     * first byte of the list, <code>1</code> being the next byte and so on. These offsets can also be
+     * negative numbers indicating offsets starting at the end of the list, with <code>-1</code> being
+     * the last byte of the list, <code>-2</code> being the penultimate, and so on.
+     *
+     * @see <a href="https://redis.io/commands/bitpos/">redis.io</a> for details.
+     * @param key The key of the string.
+     * @param bit The bit value to match.
+     * @param start The starting offset.
+     * @param end The ending offset.
+     * @return The position of the first occurrence from the <code>start</code> to the <code>end
+     *     </code> offsets of the <code>bit</code> in the binary value of the string held at <code>key
+     *     </code>.
+     * @example
+     *     <pre>{@code
+     * Long payload = client.bitpos("myKey1", 1, 4, 6).get();
+     * assert payload == 7L;// Indicates that the first occurrence of a 1 bit value starting from fifth to the sixth bytes of the string stored at "myKey1" is at the 8th position.
+     * }</pre>
+     */
+    CompletableFuture<Long> bitpos(String key, long bit, long start, long end);
+
```

Comment:
Do we need to add the following function signature as well or is that not a valid redis command? Might need to check using Redis CLI
```suggestion
CompletableFuture<Long> bitpos(String key, long bit, long start, BitmapIndexType offsetType);
```

Reply:
We have this at line 188.